interpreter/psc-interpreter.ads

```diff
@@ -1958,10 +1958,16 @@ package PSC.Interpreter is
    --  Read in the contents of the param info array
 
    --  Info on components of type
-   type Component_Info is record
+   type Component_Info (Compiled : Boolean := False) is record
       Type_Desc   : Type_Descriptor_Ptr;
       Is_By_Ref   : Boolean := False;
       Is_Optional : Boolean := True;
+      Is_Var      : Boolean := False;
+      case Compiled is
```

Comment:
I suggest you remove the "Compiled" discriminant and this variant part, and just make Decl a direct component of Component_Info.  Note that Decl itself already incorporates some degree of optionality, it being an Optional_Tree, so the "Compiled" discriminant isn't adding much here.